application/libraries/Gateways/GerencianetSdk.php

```diff
@@ -0,0 +1,120 @@
+<?php defined('BASEPATH') or exit('No direct script access allowed');
+
+use Gerencianet\Exception\GerencianetException;
+use Gerencianet\Gerencianet;
+
+class GerencianetSdk
+{
+    public function gerarBoleto(
+        $client_Id,
+        $client_Secret,
+        $nameClient,
+        $emailClient,
+        $cpfClient,
+        $phoneClient,
+        $addressRua,
+        $addressNumero,
+        $addressBairro,
+        $addressCidade,
+        $addressEstado,
+        $addressCep,
+        $idOs,
+        $title,
+        $unit_price,
+        $quantity
+    ) {
+
+        $phoneClient = preg_replace("/[^0-9]/", "", $phoneClient);
+        $addressCep = preg_replace("/[^0-9]/", "", $addressCep);
+
+        $cpfClient = preg_replace("/[^0-9]/", "", $cpfClient);
+
+        $new_unit_price = intval(preg_replace('/[^0-9]/', '', number_format($unit_price, 2, '.', '')));
+
+
+        $clientId = $client_Id; // insira seu Client_Id, conforme o ambiente (Des ou Prod)
+        $clientSecret = $client_Secret; // insira seu Client_Secret, conforme o ambiente (Des ou Prod)
+
+        $options = [
+            'client_id' => $clientId,
+            'client_secret' => $clientSecret,
+            'sandbox' => true // altere conforme o ambiente (true = desenvolvimento e false = producao)
+        ];
+
+        $item_1 = [
+            'name' => 'OS: ' . $idOs, // nome do item, produto ou serviço
+            'amount' => $quantity, // quantidade
+            'value' => $new_unit_price // valor (1000 = R$ 10,00) (Obs: É possível a criação de itens com valores negativos. Porém, o valor total da fatura deve ser superior ao valor mínimo para geração de transações.)
+        ];
+        $items = [
+            $item_1
+        ];
+        $metadata = array('notification_url' => 'http://mapos.com.br/'); //Url de notificações
```

Comment:
Se for implementado a parte de webhooks, essa parte terá que ser alterada.

Reply:
Sim, isso. Foi colocado assim porque ele não permitir passar sem metadata